plugins/utils/hardtime.nix

```diff
@@ -1,244 +1,204 @@
+        "n"
+        "x"
+      ];
+      "<CR>" = [
+        "n"
+        "x"
+      ];
+      "<C-M>" = [
+        "n"
+        "x"
+      ];
+      "<C-N>" = [
+        "n"
+        "x"
+      ];
+      "<C-P>" = [
+        "n"
+        "x"
+      ];
+    } "Keys in what modes triggering the count mechanism.";
+
+    restrictionMode =
+      defaultNullOpts.mkEnumFirstDefault
+        [
+          "block"
+          "hint"
+        ]
+        ''
+          The behavior when `restricted_keys` trigger count mechanism.
+        '';
+
+    disabledKeys = defaultNullOpts.mkAttrsOf (with types; listOf str) {
+      "<Up>" = [
+        ""
+        "i"
+      ];
+      "<Down>" = [
+        ""
+        "i"
+      ];
+      "<Left>" = [
+        ""
+        "i"
+      ];
+      "<Right>" = [
+        ""
+        "i"
+      ];
+    } "Keys in what modes are disabled.";
+
+    disabledFiletypes = defaultNullOpts.mkListOf types.str [
+      "qf"
+      "netrw"
+      "NvimTree"
+      "lazy"
+      "mason"
+    ] "`hardtime.nvim` is disabled under these filetypes.";
+
+    hints =
+      lib.nixvim.mkNullOrOption
+        (
+          with types;
+          attrsOf (submodule {
+            options = {
+              message = lib.mkOption {
+                description = "Hint message to be displayed.";
+                type = lib.nixvim.nixvimTypes.rawLua;
+              };
 
-                length = lib.mkOption {
-                  description = "The length of actual key strokes that matches this pattern.";
-                  type = types.ints.unsigned;
-                };
+              length = lib.mkOption {
+                description = "The length of actual key strokes that matches this pattern.";
+                type = types.ints.unsigned;
               };
-            })
-          )
-          ''
-            `key` is a string pattern you want to match, `value` is a table
-            of hint massage and pattern length.
-          '';
-    };
+            };
+          })
+        )
+        ''
+          `key` is a string pattern you want to match, `value` is a table
+          of hint massage and pattern length.
+        '';
   };
-
-  config =
-    let
-      setupOptions =
-        with cfg;
-        {
-          inherit
-            hint
-            notification
-            enabled
-            hints
-            ;
-
-          max_time = maxTime;
-          max_count = maxCount;
-          disable_mouse = disableMouse;
-          allow_different_key = allowDifferentKey;
-          resetting_keys = resettingKeys;
-          restricted_keys = restrictedKeys;
-          restriction_mode = restrictionMode;
-          disabled_keys = disabledKeys;
-          disabled_filetypes = disabledFiletypes;
-        }
-        // extraOptions;
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
-
-      extraConfigLua = ''
-        require("hardtime").setup(${helpers.toLuaObject setupOptions})
-      '';
-    };
 }
```

Comment:
```suggestion
settingsExample = {
// TODO: 
};
}
```

Reply:
Not a hard requirement, but it'd be great to have a useful example here :grin: 

Generally, it's best if it matches the `example` test case.